### DISCUSSION TITLE
micropython-umqtt.simple does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

### DISCUSSION THREAD

**Discussion by ElToro1966** [micropython/micropython#9874] [Libraries & Drivers] [answered]
I am not able to install several different micropython modules in a virtual environment:
```
(venv) $ python3 -m pip install micropython-umqtt.simpleCollecting micropython-umqtt.simple
  Using cached micropython-umqtt.simple-1.3.4.tar.gz (4.0 kB)
ERROR: micropython-umqtt.simple from https://files.pythonhosted.org/packages/bd/cf/697e3418b2f44222b3e848078b1e33ee76aedca9b6c2430ca1b1aec1ce1d/micropython-umqtt.simple-1.3.4.tar.gz does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```
I first tried it via Pycharm CE. Got the micropython plugin installed and when I go to packages in settings, I can see all the micropython packages I want for my project, such as umqtt.simple or ulogging, but there is always the same result when I try to install one: X does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found. Next, I tried it in the terminal as shown above. Same result.

Any help much appreciated.

**Comment by ElToro1966** [comment]
Hm... Are the micropython packages meant to be top level scripts instead of installable python-modules? micropython newbie...

**Comment by jimmo** [comment, marked as answer]
MicroPython packages aren't meant to be installed or fetched with `pip`. That would only be for installing them to your PC (but most micropython packages are not useful to run directly in CPython -- although the ones that are will generally work with `pip`).

See https://docs.micropython.org/en/latest/reference/packages.html for more info.

**Reply by ElToro1966** [reply]
Thanks.
